src/main/java/ch/njol/skript/update/skript/SkriptUpdater.java

```diff
@@ -0,0 +1,154 @@
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The updater of {@link Skript}.
+ */
+public final class SkriptUpdater extends AbstractUpdater {
+    private final Executor updateCheckerExecutor = getUpdateCheckerExecutor("Skript update checker");
+
+    private final Release current =
+            new SkriptRelease(this, Skript.getVersion(),
+                    ReleaseChannel.parseOrNightly(Skript.getVersion().toString()),
+                    ReleaseStatus.CURRENT, true);
+
+    public SkriptUpdater(final boolean enabled,
+                         final ReleaseChannel channel,
+                         final long frequency,
+                         final TimeUnit unit) {
+        super(enabled, channel, frequency, unit);
+    }
+
+    @Override
+    public void setAutoInstallEnabled(final boolean enabled) {
+        SkriptConfig.automaticallyDownloadNewVersion.setValue(enabled);
+    }
+
+    @Override
+    public boolean isAutoInstallEnabled() {
+        return SkriptConfig.automaticallyDownloadNewVersion.value();
+    }
+
+    @Override
+    public String getName() {
+        return "Skript";
+    }
+
+    @Override
+    public Release getCurrentRelease() {
+        return current;
+    }
+
+    @Override
+    public CompletableFuture<Release> getLatestRelease() {
+        state = UpdaterState.CHECKING;
+        return CompletableFuture.supplyAsync(() -> {
+            final String latestVersion;
+            if (Skript.latestVersion != null)
+                latestVersion = Skript.latestVersion;
+            else {
+                assert false : Skript.getVersionWithSuffix();
+                latestVersion = Skript.getVersion().toString();
+            }
+            state = UpdaterState.PENDING_DOWNLOAD;
+            return new SkriptRelease(SkriptUpdater.this, new Version(latestVersion), SkriptUpdater.this.channel, ReleaseStatus.LATEST, false);
+        }, updateCheckerExecutor);
+    }
+
+    @Override
+    public void installRelease(final Release release) throws InstalledReleaseException, IOException {
+        if (release.isInstalled())
+            throw new InstalledReleaseException("Release is already installed");
+        if (state == UpdaterState.DOWNLOADING || state == UpdaterState.INSTALLING || state == UpdaterState.PENDING_RESTART)
+            throw new InstalledReleaseException("A release is already installed");
+        state = UpdaterState.DOWNLOADING;
+
+        current.backup();
+
+        final File serverDirectory = Skript.getInstance().getDataFolder().getParentFile().getCanonicalFile().getParentFile().getCanonicalFile();
+        final File updatedJar = Paths.get(serverDirectory.getCanonicalPath(), "plugins", "Skript", "update", "Skript.jar").toFile();
+
+        updatedJar.getParentFile().mkdirs();
+        updatedJar.createNewFile();
+
+        try (final InputStream stream = Skript.invoke(Skript.urlOf(release.getDownloadUrl()).openConnection(), connection -> {
```

Comment:
Maybe implement a common method for downloading and replacing JAR files in runtime?